carbon-route.html

```diff
@@ -345,9 +359,35 @@
         return value;
       }, this);
       if (values.tail && values.tail.path) {
-        interp.push(values.tail.path);
+        if (interp.length > 0 && values.tail.path.charAt(0) === '/') {
+          interp.push(values.tail.path.slice(1));
+        } else {
+          interp.push(values.tail.path);
+        }
       }
       return interp.join('/');
+    },
+
+    __setMulti(setObj) {
+      // HACK(rictic): skirting around 1.0's lack of a setMulti by poking at
+      //     internal data structures. I would not advise that you copy this
+      //     example.
+      //
+      //     In the future this will be a feature of Polymer itself.
+      //     See: https://github.com/Polymer/polymer/issues/3640
+      //
+      //     Hacking around with private methods like this is juggling footguns,
+      //     and is likely to have unexpected and unsupported rough edges.
+      //
+      //     Be ye so warned.
+      for (var property in setObj) {
+        this._propertySetter(property, setObj[property]);
+      }
+
+      for (var property in setObj) {
+        this._pathEffector(property, this[property]);
+        this._notifyPathUp(property, this[property]);
+      }
```

Comment:
Can you expand on your comment? Is there a better way to achieve this effect?

The issue is due to the cyclical flow of effects between three properties on carbon-route: `path` (technically `route.path`), `data`, and `tail`. `tail` and `data` are implied by `path`, but likewise `path` is implied by `tail` and `data`. Normally this doesn't cause an infinite loop because of dirty checking, we update `path` which updates `tail` and `data`, when we process the changes to `tail` and `data` to produce a new `path` it is the same as the existing path and so the updates stop there.

The issue is that for this to work robustly we need to update `data` and `tail` atomically. That is, all observers of `tail` and `data` must see a consistent view of both, so we need to separate out updating `__data__` with processing effects and notifying up.

If there were a method in Polymer core to do this I would gladly use it, and I've proposed one at https://github.com/Polymer/polymer/issues/3640